marketplace_builder/views/pages/how-to/e-commerce/configuring-data-schema.liquid

```diff
@@ -0,0 +1,225 @@
+---
+converter: markdown
+metadata:
+  title: Configuring Data Schema
+  description: Tutorial on how to build e-commerce sites powered by PlatformOS
+slug: how-to/e-commerce/configuring-data-schema
+searchable: true
+---
+
+{% include 'how-to/e-commerce/tutorial-info' %}
+
+In PlatformOS we are pretty much free to build any kind of data model we want. Defining data schema is very similar to creating any other relational database. We can use one of the many tools to create a visual representation of the model. We’ve gone with [dbdiagram.io](https://dbdiagram.io/).
+
+{% include 'how-to/e-commerce/data-model-graph' %}
+
+Our data schema will be based on three primary models - `user`, `product` and `order`. On PlatformOS side we will use `User` to map the corresponding model and `Custom Model Type` to represent everything else.
+
+## Product
+
+Every t-shirt in our store will be described by `product_type` and `brand`. Additionally every product may have one or more `photo` attached. Finally, there is a mater of available variants of the product (in our case it’s t—shirt sizes). We will create `stock_level` tht will hold information on the available amount of products in specific variant (size).
+
+```yaml
+# custom_model_types/product.yml
+name: product
+custom_attributes:
+  - name: name
+    attribute_type: string
+  - name: description
+    attribute_type: text
+  - name: sex
+    attribute_type: string
+  - name: price
+    attribute_type: int
+  - name: brand_id
+    attribute_type: string
+  - name: product_type_id
+    attribute_type: string
+```
+
+Few items worth noting:
+
+- PlatformOS does not support `enum` data type at the moment. We will handle that on the frontend side with constants.
+- Price will stored as an integer, representing cents, so \$1.00 is stored as `100`. This helps us to avoid problematic computations and rounding problems that sometime happen with `float` data type.
+- External ids are integer numbers but since variable types are important and `"100" != 100` it’s easier to store all IDs as strings. We will talk more on this problem in later chapters.
+
+```yaml
+# custom_model_types/brand.yml
+name: brand
+custom_attributes:
+  - name: name
+    attribute_type: string
+```
+
+```yaml
+# custom_model_types/product_type.yml
+name: product_type
+custom_attributes:
+  - name: name
+    attribute_type: string
+```
+
+```yaml
+# custom_model_types/photo.yml
+name: photo
+custom_attributes:
+  - name: product_id
+    attribute_type: string
+  - name: image
+    attribute_type: image
+    attribute_type: photo
+    versions_configuration:
```

Comment:
i suggest  we simplify the example to one image version and use direct s3 upload to avoid promoting deprecated methods

Reply:
I would rather keep the example complete as it’s in the source code, so I will keep the versions.

I’ve removed duplicated and invalid `attribute_type: image`

Reply:
Alright, if it’s gone then it should probably throw an error during sync (I am developing on new stack and was not aware of this).

I can get rid of `versions_configuration` at this stage and add it later on.

Reply:
cc @piotrze - looks like we are missing an error message ^

Reply:
@pavelloz Haven't you added this `versions_configuration` few days/weeks/months ago?

Reply:
Nah, Maciek added `image_versions` for future lambda support and its forwarding metadata to form_builder only